website/content/docs/intro/use-cases/blue-green-deployment.mdx

```diff
@@ -0,0 +1,447 @@
+      ]
+    }
+  }
+}
+
+// omitted other stanzas for brevity
+  deploy {
+    use "nomad-jobspec" {
+      jobspec = templatefile("${path.app}/nomad/job.nomad.tpl", {
+        // The registry username and password are passed to the Nomad job template
+        // so that the Nomad client can pull the image from the Docker registry.
+        username = var.username
+        password = var.password
+      })
+    }
+  }
+//
+```
+
+The `split-traffic-to-green` step in this pipeline first downloads a file from the
+git repository to the container performing the exec step of this pipeline using `curl`.
+Then, since it is running the official Consul Docker image, the command `consul config write` is executed, which writes the downloaded configuration entry to the Consul
+cluster via the API. The configuration entry looks like the below:
+
+```hcl
+Kind= "service-splitter"
+Name = "app"
+Splits = [
+  {
+    Weight = 50
+    ServiceSubset = "blue"
+  },
+  {
+    Weight = 50
+    ServiceSubset= "green"
+  }
+]
+```
+
+This is a service splitter config entry. The different "subsets" in this configuration,
+`blue`, and `green`, align with a service resolver that has already been configured
+in the Consul cluster. This configuration (as a [Terraform resource](https://github.com/paladin-devops/hashitalks-deploy-blue-green/blob/main/terraform/main.tf#L14-L31)
+using the Consul provider) is depicted below, where instances of the "app" service
+tagged with `blue` are a part of the `blue` subset, and those tagged with `green` are
+a part of the `green` subset:
+
+```terraform
+resource "consul_config_entry" "app_service_resolver" {
+  name = "app"
+  kind = "service-resolver"
+
+  config_json = jsonencode({
+    Kind          = "service-resolver"
+    Name          = "app"
+    DefaultSubset = "blue"
+    Subsets = {
+      blue = {
+        Filter = "blue in Service.Tags"
+      }
+      green = {
+        Filter = "green in Service.Tags"
+      }
+    }
+  })
+}
+```
+
+Since a Nomad job is being deployed in this pipeline, the tags on the service instances
+are derived from the Nomad jobspec template. Specifically, this is configured in the
+`service` stanza of the Nomad job. The `tags` are applied to the Nomad allocations
+running the current active version of the Nomad job, and `canary_tags` are applied
+to the canary Nomad allocations rolled out during a deployment.
+
+```hcl
+    service {
+      name        = "app"
+      port        = "http"
+      tags        = [ "blue" ]
+      canary_tags = [ "green" ]
+    }
+```
+
+The exact number of canary Nomad allocations to be deployed, with the `canary_tags`
+from the `service` stanza, is configured in the `update` stanza's `canary` field:
+
+```hcl
+    update {
+      max_parallel = 1
+      canary       = 1
+      auto_revert  = true
+      auto_promote = false
+      health_check = "task_states"
+    }
```

Comment:
Left align this I think and it will probably look nicer in the rendered code block!